butterfree/validations/basic_validaton.py

```diff
@@ -1,7 +1,8 @@
 """Validation implementing basic checks over the dataframe."""
 
-from typing import Optional
+from typing import Optional, Union
 
+from pyspark.sql.connect.dataframe import DataFrame as ConnectDataFrame
```

Comment:
minor: wouldn't this break for clients using butterfree below 3.4 without connect support?

Reply:
Added type checking